### ISSUE TITLE
aws_elasticache_replication_group does not provide cache_nodes output

### ISSUE THREAD

**Issue by hashibot** [terraform-providers/terraform-provider-aws#266]
_This issue was originally opened by @brikis98 as hashicorp/terraform#8788. It was migrated here as part of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

### Terraform Version

Terraform v0.7.3
### Affected Resource(s)
- aws_elasticache_replication_group

The `aws_elasticache_replication_group` resource only provides two outputs: `id` and `primary_endpoint_address`. But what about the IDs and addresses of each node in the cluster?

The `elasticache_cluster` resource provides a `cache_nodes` output, which provides the `id`, `address`, `port` and `availability_zone` of every node in the cluster. Is there a reason `aws_elasticache_replication_group` does not do the same? We need this data to use replication groups with Redis and we cannot get them from the `elasticache_cluster` resource, as that only creates single-node redis clusters.

Also, the ID of the underlying cluster is not available as an output either. The `id` output is the ID of the replication group (e.g. `foo`), but the underlying cluster will have a different id (e.g. `foo-001`).

**Comment by Alberts00** [comment]
Hey @kwilczynski
Are you still planning on updating the output?

**Comment by kwilczynski** [comment]
Hi @Alberts00!

I have to admit that I have forgotten about this due to things happening in my life at the point (moving countries, etc.). I will have a look again.

Apologies!

**Comment by davideme** [comment]
Any news about this? How can we help ?

**Comment by davebwag** [comment]
This would be a really good one... any word on this?

**Comment by bflad** [comment]
The `aws_elasticache_replication_group` data source and resource will soon provide a new attribute, `member_clusters` (matching `MemberClusters` from the ElastiCache API), with IDs of all underlying clusters within a replication group. This will be released with version 1.26.0 of the AWS provider shortly.

**Comment by pmcevoy** [comment]
`cache_nodes` output would have been better as it provides more information.  If I want to get the endpoint addresses of all the nodes in the replication group, I have to use this hack (after the group has been created in a different module):

```
variable "cluster_name" {
}
variable "dns_zones" {
  type = "list"
}

data "aws_elasticache_replication_group" "members" {
  replication_group_id = "${var.cluster_name}"
}
data "aws_elasticache_cluster" "cluster" {
  count = "${length(var.dns_zones)}"
  cluster_id = "${data.aws_elasticache_replication_group.members.member_clusters[count.index]}"
}

output "addresses" {
  value = ["${data.aws_elasticache_cluster.cluster.*.cache_nodes.0.address}"]
}
```
